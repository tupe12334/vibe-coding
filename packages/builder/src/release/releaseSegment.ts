import type { ListItem, RootContent } from "mdast";
import { releaseItRules } from "./release-it";
import { semanticReleaseRules } from "./semantic-release";

const releaseRulesMap: Record<string, readonly string[]> = {
  "release-it": releaseItRules,
  "semantic-release": semanticReleaseRules,
};

export const releaseSegment = async (
  releaseSystem: string
): Promise<RootContent[]> => {
  const releaseItems = releaseRulesMap[releaseSystem] ?? [];

  const segment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [
        { type: "text", value: "Release system" },
        { type: "text", value: ` (${releaseSystem})` },
      ],
    },
    {
      type: "list",
      ordered: false,
      children: releaseItems.map(
        (item): ListItem => ({
          type: "listItem",
          children: [
            { type: "paragraph", children: [{ type: "text", value: item }] },
          ],
        })
      ),
    },
  ];

  return segment;
};
