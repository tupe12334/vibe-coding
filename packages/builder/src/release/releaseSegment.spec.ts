import { describe, expect, test } from "vitest";
import { releaseSegment } from "./releaseSegment";

describe("releaseSegment", () => {
  test("returns release system guidelines", async () => {
    const segment = await releaseSegment("release-it");
    expect(segment).toMatchSnapshot();
  });
});
