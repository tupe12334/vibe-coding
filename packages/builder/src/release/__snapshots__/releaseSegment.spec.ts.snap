// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`releaseSegment > returns release system guidelines 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "Release system",
      },
      {
        "type": "text",
        "value": " (release-it)",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [
              {
                "type": "text",
                "value": "Use release-it for interactive version management and publishing.",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "ordered": false,
    "type": "list",
  },
]
`;
